extensions/mssql/src/errorDiagnostics/errorDiagnosticsProvider.ts

```diff
@@ -68,14 +73,52 @@ export class ErrorDiagnosticsProvider extends SqlOpsFeature<any> {
 				});
 			}
 
+			private async showCertValidationDialog(connection: azdata.IConnectionProfile, errorMessage: string, callStack: string): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> {
+				try {
+					let actions: azdata.window.IDialogAction[] = [];
+					let trustServerCertAction: azdata.window.IDialogAction = {
+						id: ErrorDiagnosticsConstants.TSC_ActionId,
+						label: ErrorDiagnosticsConstants.TSC_EnableTrustServerCert,
+						isEnabled: true,
+						isPrimary: true,
+						closeDialog: true
+					};
+
+					actions.push(trustServerCertAction);
+					const result = await azdata.connection.openCustomErrorDialog(
+						{
+							severity: azdata.window.MessageLevel.Error,
+							headerTitle: ErrorDiagnosticsConstants.ConnectionErrorDialogTitle,
+							message: errorMessage,
+							messageDetails: callStack,
+							instructionText: ErrorDiagnosticsConstants.TSC_InstructionText,
+							readMoreLink: ErrorDiagnosticsConstants.TSC_ReadMoreLink,
+							actions: actions
+						}
+					);
+
+					// Result represents id of action taken by user.
+					if (result && result === ErrorDiagnosticsConstants.TSC_ActionId) {
```

Comment:
nit: Falsy values will never strict equal a non-empty string
```suggestion
					if (result === ErrorDiagnosticsConstants.TSC_ActionId) {
```